spec/API_specification/creation_functions.md

```diff
@@ -49,6 +49,42 @@ This function cannot guarantee that the interval does not include the `stop` val
 
     -   a one-dimensional array containing evenly spaced values. The length of the output array must be `ceil((stop-start)/step)`.
 
+
+(function-asarray)=
+### asarray(obj, /, *, dtype=None, device=None, copy=False)
+
+Convert the input to an array.
+
+#### Parameters
+
+-   **obj**: _Union\[ float, NestedSequence\[ bool | int | float ], SupportsDLPack, SupportsBufferProtocol, memoryview ]_
+
+    -   Object to be converted to an array. Can be a Python scalar, a (possibly nested) sequence of Python scalars, or an object supporting DLPack or the Python buffer protocol.
+
+    :::{tip}
+    An object supporting DLPack has `__dlpack__` and `__dlpack_device__` methods.
+    An object supporting the buffer protocol can be turned into a memoryview through `memoryview(obj)`.
+    :::
+
+-   **dtype**: _Optional\[ &lt;dtype&gt; ]_
+
+    -   output array data type. If `dtype` is `None`, the output array data type must be inferred from the data type(s) in `obj`. Default: `None`.
+
+-   **device**: _Optional\[ &lt;device&gt; ]_
+
+    -   device to place the created array on, if given. Default: `None`.
+
+-   **copy**: _bool_
+
+    -   Whether or not to make a copy of the input. If `True`, always copies. If `False`, reuses existing memory buffer if possible. Default: `False`.
```

Comment:
We should strongly consider revising the behavior of the `copy` argument from NumPy so that `copy=False` means _never_ copy. In other words:

- `copy=None` (default): copy only if necessary.
- `copy=True`: always copy.
- `copy=False` : never copy. Raise `ValueError` if a copy would be needed.

Reply:
Yes, that's a very good point. We never got around to it in NumPy, but here we should fix it.

Reply:
Also added this to the NEP as something that must be improved outside of the `array_api` namespace.